maxi-blocks/maxi-blocks
# Issue 3558 - SVG transition support

# Description

Adds transition support for colour and background on SVG Icon block. Also fixes:
1. An error while trying to change the SVG colour on IB:

https://user-images.githubusercontent.com/50477222/184335486-9cb0215f-f63c-4420-ac26-8f2283574364.mp4

2. An error with the transition effects when using IB on SVG Icon block, specifically for elements without `data-fill` or `data-stroke` attributes, like happens on this `circle` of this [code editor example](https://github.com/maxi-blocks/maxi-blocks/files/9315585/ibcircletransition.txt):

https://user-images.githubusercontent.com/50477222/184335666-ff30a810-7d99-451a-9a23-0e9a60406977.mp4

3. Optimize a bit the code on `src/extensions/styles/helpers/getSVGStyles.js`

Fixes #3553 

# How Has This Been Tested?

1. Try to change the colour of the SVG Icon from IB
2. Try the effect on the frontend: the circle should go smoothly as the rest of the icon
3. On an SVG Icon block, on adding hover colour and background, both should have transition options 👍 

# Test checklist



**_ Front/Back Testing _**


**_ Pre-Code Testing _**


# Checklist
